packages/starlight/__tests__/test-config.ts

```diff
@@ -4,11 +4,19 @@ import { getViteConfig } from 'astro/config';
 import type { z } from 'astro/zod';
 import { vitePluginStarlightUserConfig } from '../integrations/virtual-user-config';
 import { StarlightConfigSchema } from '../utils/user-config';
+import type { AstroConfig } from 'astro';
+
+export function defineVitestConfig(
+	config: z.input<typeof StarlightConfigSchema>,
+	opts?: Partial<AstroConfig>
```

Comment:
And then same type here should avoid the assertions in test config:

```suggestion
	opts?: { build: Pick<AstroConfig['build'], 'format'> }
```